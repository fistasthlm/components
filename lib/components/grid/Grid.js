import React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import BikeTile from './tile/Tile'
import style from './Grid.scss'

export default function Grid ({bikes}) {
  return (
    <div className={style.grid}>
      {
        bikes.size > 0 &&
        bikes.map(bike => {
          return (
            <BikeTile
              key={bike.get('_id')}
              bike={bike} />
          )
        })
      }
    </div>
  )
}

Grid.propTypes = {
  bikes: ImmutablePropTypes.list.isRequired
}
