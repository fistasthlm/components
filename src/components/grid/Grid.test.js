import React from 'react';
import { shallow } from 'enzyme';
import { Map, List } from 'immutable';
import Grid from './Grid';

const setup = propOverrides => {
  const props = Object.assign({
    bikes: List(),
  }, propOverrides);

  return shallow(<Grid {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render grid', () => {
    const wrapper = setup({
      bikes: List([
        Map({
          _id: 'yolo',
          title: 'dolan',
        })
      ])
    });

    expect(wrapper).toMatchSnapshot();
  });
});
