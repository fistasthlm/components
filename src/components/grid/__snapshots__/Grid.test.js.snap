// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `<div />`;

exports[`Render should render grid 1`] = `
<div>
  <Tile
    bike={
      Immutable.Map {
        "_id": "yolo",
        "title": "dolan",
      }
    }
    key="yolo"
  />
</div>
`;
